### PR TITLE
add documentation for `lifecycle hooks`

### PR DESCRIPTION
**Pull request by gsquared94**
Related: #1441
http://35.236.30.3:1313/docs/pipeline-stages/lifecycle-hooks/

### CODE REVIEW

**Comment by MarlonGamez** [comment]
For some reason when looking at the preview docs there is a bug with the footer not being at the bottom of the page 
Screen Shot 2021-07-29 at 10 28 45 AM

I'm not sure if this is somehow an issue with the PR, but it seems to only happen on the lifecycle hooks page

File: docs/content/en/docs/pipeline-stages/lifecycle-hooks.md [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+$SKAFFOLD_RUN_ID | Run specific UUID label for deployed or to be deployed resources | Deploy
+$SKAFFOLD_DEFAULT_REPO | The resolved default repository | All
+$SKAFFOLD_RPC_PORT | TCP port to expose event API | All
+$SKAFFOLD_HTTP_PORT | TCP port to expose event REST API over HTTP | All
+$SKAFFOLD_KUBE_CONTEXT | The resolved Kubernetes context | Sync, Deploy
+$SKAFFOLD_NAMESPACES | Comma separated list of Kubernetes namespaces | Sync, Deploy
+$SKAFFOLD_WORK_DIR | The workspace root directory | All
+Local environment variables | The current state of the local environment (e.g. $HOST, $PATH). Determined by the golang os.Environ function. | All
+
+</div><br /></b>
```
Comment (MarlonGamez) [inline docs/content/en/docs/pipeline-stages/lifecycle-hooks.md:91]: I think this is causing the issue with the page footer. Should it be here? Removing it locally seems to fix it with no side effects but wasn't sure if this was something intentionally added

File: docs/content/en/docs/pipeline-stages/lifecycle-hooks.md [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+$SKAFFOLD_FILES_ADDED_OR_MODIFIED | Semi-colon delimited list of absolute path to all files synced or to be synced in current dev loop that have been added or modified | Sync
+$SKAFFOLD_FILES_DELETED | Semi-colon delimited list of absolute path to all files synced or to be synced in current dev loop that have been deleted | Sync
+$SKAFFOLD_RUN_ID | Run specific UUID label for deployed or to be deployed resources | Deploy
+$SKAFFOLD_DEFAULT_REPO | The resolved default repository | All
+$SKAFFOLD_RPC_PORT | TCP port to expose event API | All
+$SKAFFOLD_HTTP_PORT | TCP port to expose event REST API over HTTP | All
+$SKAFFOLD_KUBE_CONTEXT | The resolved Kubernetes context | Sync, Deploy
+$SKAFFOLD_NAMESPACES | Comma separated list of Kubernetes namespaces | Sync, Deploy
+$SKAFFOLD_WORK_DIR | The workspace root directory | All
+Local environment variables | The current state of the local environment (e.g. $HOST, $PATH). Determined by the golang os.Environ function. | All
```
Comment (MarlonGamez) [inline docs/content/en/docs/pipeline-stages/lifecycle-hooks.md:89]: Should this line be here?

**Review by MarlonGamez — COMMENTED** [review]
Looking good, just have a couple small questions regarding some lines

File: docs/content/en/docs/pipeline-stages/lifecycle-hooks.md (hunk 1 above)
Comment (aaron-prindle) [reply docs/content/en/docs/pipeline-stages/lifecycle-hooks.md:91]: What is wrong w/ the footer?
image

**Review by aaron-prindle — APPROVED** [review]
LGTM

File: docs/content/en/docs/pipeline-stages/lifecycle-hooks.md (hunk 1 above)
Comment (MarlonGamez) [reply docs/content/en/docs/pipeline-stages/lifecycle-hooks.md:91]: oh, do you not experience the issue in my previous comment? When running it locally if I remove the html on line 91 it seems to fix it

File: docs/content/en/docs/pipeline-stages/lifecycle-hooks.md (hunk 2 above)
Comment (gsquared94) [reply docs/content/en/docs/pipeline-stages/lifecycle-hooks.md:89]: you're right. I think that's what was causing the weird scroll behavior also

File: docs/content/en/docs/pipeline-stages/lifecycle-hooks.md (hunk 1 above)
Comment (gsquared94) [reply docs/content/en/docs/pipeline-stages/lifecycle-hooks.md]: fixed in https://github.com/GoogleContainerTools/skaffold/pull/6330#discussion_r679631867

**Pull request merged**